### PR TITLE
Register as a Restaurant Owner

### PR DESCRIPTION
**Pull request by tochman**
PT Story: https://www.pivotaltracker.com/story/show/131080433

Changes proposed in this pull request:
- adds scopes for `customers` and `owners`
- adds role as sign up param
- modifies `registrations#new` method to allow for different roles
- modifies sign up form to include hidden field for role
- adds validation for roles 
- adds `owner?` method to User 
- adds authorization engine with basic setup
- adds `Restaurant` class with association to user (restricted to `owner`)
- adds placeholder view for new restaurant form
- updates cart view

What I have learned working on this feature: 
- that the Rails way to create objects is sometimes hard to understand ;-)
- that sometimes it is still hard to stop once you are in a flow and how easy it is to get out of scope with a feature and deliver more than you should.

**Ready for review**

### CODE REVIEW

File: .idea/.generators [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AmberWilkie) [inline .idea/.generators]: Is my seeing this file here mean it is not actually removed from version control?

File: .idea/.name [hunk 2]
```diff
@@ -0,0 +1 @@
+sf-online-august
```
Comment (AmberWilkie) [inline .idea/.name]: same

File: .idea/.rakeTasks [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AmberWilkie) [inline .idea/.rakeTasks]: same

File: config/routes.rb [hunk 4]
```diff
@@ -1,9 +1,10 @@
 Rails.application.routes.draw do
 
-  get '/restaurant', controller: :restaurants, action: :index
+  #get '/restaurant', controller: :restaurants, action: :index
```
Comment (AmberWilkie) [inline config/routes.rb]: ???

File: db/migrate/20161003033041_create_restaurants.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class CreateRestaurants < ActiveRecord::Migration[5.0]
+  def change
+    create_table :restaurants do |t|
+      t.string :name
+      t.text :description
+      t.references :user, foreign_key: true
+
```
Comment (AmberWilkie) [inline db/migrate/20161003033041_create_restaurants.rb:7]: Restaurants need an address. Were you planning to add that in a later feature?

**Review by AmberWilkie — COMMENTED** [review]
I more or less understand what all this code is doing, though it's a lot to take in and new techniques and gems. I think it would be helpful to take 15 minutes to explain again what CanCan is doing, how Abilities work and what Scope is about.

File: db/migrate/20161003033041_create_restaurants.rb (hunk 5 above)
Comment (tochman) [reply db/migrate/20161003033041_create_restaurants.rb:7]: Not part of this story. I had no requirements to follow.

File: config/routes.rb (hunk 4 above)
Comment (tochman) [reply config/routes.rb]: Good catch. Will remove

File: .idea/.generators (hunk 1 above)
Comment (tochman) [reply .idea/.generators]: Is it still part of the PR? Will investigate

File: db/migrate/20161003033041_create_restaurants.rb (hunk 5 above)
Comment (AmberWilkie) [reply db/migrate/20161003033041_create_restaurants.rb:7]: Alright, fair enough. (I guess. You totally changed the story before working on the feature...)

File: .idea/.generators (hunk 1 above)
Comment (tochman) [reply .idea/.generators]: removed now. Mea Culpa

File: config/routes.rb (hunk 4 above)
Comment (tochman) [reply config/routes.rb]: Done

**Pull request merged**